src/control/server/server.go

```diff
@@ -187,11 +185,14 @@ func Start(log *logging.LeveledLogger, cfg *Configuration) error {
 	}
 	defer netdetect.CleanUp(netCtx)
 
-	for i, srvCfg := range cfg.Servers {
-		if i+1 > maxIOServers {
-			break
-		}
+	// On a NUMA-aware system, emit a message when the configuration
+	// may be sub-optimal.
+	numaCount := netdetect.NumNumaNodes(netCtx)
+	if numaCount > 0 && len(cfg.Servers) > numaCount {
+		log.Infof("NOTICE: Detected %d NUMA node(s); %d-server config may not perform as expected", numaCount, len(cfg.Servers))
```

Comment:
would suggest changing from "may not perform as expected" to "will not utilise all available resources"

Reply:
This message is printed when the number of configured servers is greater than the number of detected NUMA nodes, when the number of detected NUMA nodes is greater than zero. The reasoning is:
- If there are zero NUMA nodes detected (common on VMs and non-production systems), then this message would just be spam, so don't print it
- If the number of configured servers is greater than the number of detected NUMA nodes, then we should print this message to let the admin know that they are probably oversubscribing the hardware
- If the number of configured servers is less than the number of detected NUMA nodes, we don't currently print a message because this may be a deliberate choice and we don't want to spam the console

I did consider a message for the last item, but decided against it for the moment because we need to be sensitive about excessive logging. As we develop more auto-config features I think we can do more to help the admin arrive at an optimal config for the hardware.